pkg/ibmvpc/nacl_analysis.go

```diff
@@ -468,60 +499,68 @@ func (na *NACLAnalyzer) AllowedConnectivity(subnetCidr, inSubentCidr, target str
 // if conn is specified then rules contributing to that connection; otherwise to any connection src->dst
 // if the input subnet was not yet analyzed, it first adds its analysis to saved results
 func (na *NACLAnalyzer) rulesInConnectivity(subnetCidr, inSubentCidr,
-	target string, connQuery *common.ConnectionSet, isIngress bool) ([]int, error) {
+	target string, connQuery *common.ConnectionSet,
+	isIngress bool) (allow, deny []int, err error) {
 	// add analysis of the given subnet
 	// analyzes per subnet disjoint cidrs (it is not necessarily entire subnet cidr)
 	analyzedConns, targetIPblock, inSubnetIPblock := na.initConnectivityRelatedCompute(subnetCidr, inSubentCidr, target, isIngress)
 
 	for disjointSubnetCidr, analyzedConnsPerCidr := range analyzedConns {
 		disjointSubnetCidrIPblock, err := common.IPBlockFromIPRangeStr(disjointSubnetCidr)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		if inSubnetIPblock.ContainedIn(disjointSubnetCidrIPblock) {
-			for resTarget, rules := range analyzedConnsPerCidr.contribRules {
+			for resTarget, allowRules := range analyzedConnsPerCidr.allowRules {
 				if !targetIPblock.ContainedIn(resTarget) {
 					continue
 				}
+				// this is the relevant targetIPblock; takes denyRules as well
+				denyRules := analyzedConnsPerCidr.denyRules[resTarget]
 				if connQuery == nil {
-					return rules, nil
+					return allowRules, denyRules, nil
 				}
+				var mergedRules, allowRelevant, denyRelevant []int
+				mergedRules = append(mergedRules, allowRules...)
+				mergedRules = append(mergedRules, denyRules...)
```

Comment:
```suggestion
				mergedRules = append(allowRules, denyRules...)
```

Reply:
Its only from go.1.22, I have go.1.21 installed.
Lets deffer this to the time we decide to update to go.1.22 across the broad; I'll add todo